components/brave_wallet/browser/internal/orchard_storage/orchard_storage.cc

```diff
@@ -1409,10 +1405,10 @@ OrchardStorage::RemoveCheckpoint(const mojom::AccountIdPtr& account_id,
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
   CHECK(account_id);
 
-  auto existing_checkpoint = GetCheckpoint(account_id, checkpoint_id);
-  RETURN_IF_ERROR(existing_checkpoint);
+  ASSIGN_OR_RETURN(auto existing_checkpoint,
+                   GetCheckpoint(account_id, checkpoint_id));
 
-  if (!existing_checkpoint.value()) {
+  if (!existing_checkpoint.has_value()) {
     return base::ok(Result::kNone);
```

Comment:
`  if (!existing_checkpoint.value()) {`
as existing_checkpoint is not in a error state, but has empty optional payload

Reply:
or it underlying type with this new macros?